### DISCUSSION TITLE
Exposing Local Instance to Port

### DISCUSSION THREAD

**Discussion by ngriffiths13** [geldata/gel#3754] [Q&A] [answered]
I have been experimenting with EdgeDB for the past month or so now. It has been amazing. I like it quite a bit more than the Postgres I am used to. I created a DB in the recommended way for development in a project and have been using it there. I now want to take that DB and expose it so other machines could access the DB also. I have followed the instructions on creating roles, but I am having a hard time understanding how to now allow other machines to connect to my DB. I tried to connect using the DSN, but it seems to not like that I am passing an IP address instead of a hostname. 

I was hoping this would be as simple as a few firewall rules and a connection string, but I am not having much luck. Are there any good tutorials out there for how to set up a remote DB using EdgeDB and connect to it?

**Comment by elprans** [comment, marked as answer]
> but it seems to not like that I am passing an IP address instead of a hostname

Can you elaborate on this please?  What error are you seeing?

**Reply by ngriffiths13** [reply]
The exact error is this:
`edgedb error: ClientError: Cannot use verify_hostname or system root certificates with an IP address`

**Reply by ngriffiths13** [reply]
I am trying to connect using the following command:

`edgedb --dsn edgedb://myuser:password@ip_address:port/db_name`

**Reply by elprans** [reply]
Try `--tls-security=insecure` to disable TLS host verification.

**Reply by ngriffiths13** [reply]
Do the listen ports need to be updated to be expecting connections from outside machines?

**Reply by ngriffiths13** [reply]
I am also a little confused why the documentation says that the default port that edgedb listens to for connection is 5656 but the DB instance defaults to running on 10700. Which one is the right one to connect to?

**Reply by elprans** [reply]
> default port that edgedb listens to for connection is 5656

This is when you run `edgedb-server` (or the Docker container) directly.  Local development instances created with `edgedb project init` and `edgedb instance create` run on arbitrary ports, and aren't really intended to be accessed from other machines.  If you want a "proper" deployment that is available to other hosts, there is a number of guides here: https://www.edgedb.com/docs/guides/deployment/index

**Reply by ngriffiths13** [reply]
Ah okay so this is what I was looking for. Sorry for the confusion. And it looks fairly trivial to dump and load data from my dev db to a production one. Thank you for answering my questions!

**Reply by ngriffiths13** [reply]
@elprans Could you give me the string that would be needed to connect to the deployed docker container as shown in the example? I am getting connection denied errors when trying to connect using the following:
`edgedb --dsn edgedb://edgedb@localhost:5656/edgedb`

For reference the docker-compose:
```version: "3"
services:
  edgedb:
    image: edgedb/edgedb
    environment:
      EDGEDB_SERVER_SECURITY: insecure_dev_mode
    volumes:
      - "./dbschema:/dbschema"
      - "./credentials:/root/.config/edgedb/credentials"
    ports:
      - "5656"```

**Reply by elprans** [reply]
Docker proxies on localhost only by default, try:

```
ports:
   - 0.0.0.0:5656:5656
```
